vowpalwabbit/options.h

```diff
@@ -88,25 +91,36 @@ struct typed_option : base_option
   typed_option& value(T value)
   {
     m_value = std::make_shared<T>(value);
+    if (m_location != nullptr) { *m_location = value; }
+    value_set_callback(value);
     return *this;
   }
 
   T value() const { return m_value ? *m_value : T(); }
 
-  T& m_location;
+protected:
+  // Allows inheriting classes to handle set values. Noop by default.
+  virtual void value_set_callback(T /*value*/) {}

```

Comment:
should T be const, in case its a list? or that doesnt make sense here

Reply:
Const ref makes sense yeah